src/hotspot/share/opto/superword.hpp

```diff
@@ -209,56 +207,89 @@ class SuperWord : public ResourceObj {
-  PhaseIterGVN& igvn()        const { return vloop().phase()->igvn(); }
-  IdealLoopTree* lpt()        const { return vloop().lpt(); }
-  CountedLoopNode* cl()       const { return vloop().cl(); }
-  PhiNode* iv()               const { return vloop().iv(); }
-  int iv_stride()             const { return cl()->stride_con(); }
-  bool in_bb(const Node* n)   const { return vloop().in_bb(n); }
+  const VLoop& vloop()                  const { return vloop_analyzer().vloop(); }
+  PhaseIdealLoop* phase()               const { return vloop().phase(); }
+  PhaseIterGVN& igvn()                  const { return vloop().phase()->igvn(); }
+  IdealLoopTree* lpt()                  const { return vloop().lpt(); }
+  CountedLoopNode* cl()                 const { return vloop().cl(); }
+  PhiNode* iv()                         const { return vloop().iv(); }
+  int iv_stride()                       const { return cl()->stride_con(); }
+  bool in_bb(const Node* n)             const { return vloop().in_bb(n); }
+
+  // VLoopReductions Accessors
+  bool is_marked_reduction(const Node* n) const {
+    return vloop_analyzer().reductions().is_marked_reduction(n);
+  }
 
-#ifndef PRODUCT
-  // TraceAutoVectorization and TraceSuperWord
-  bool is_trace_superword_vector_element_type() const {
-    // Too verbose for TraceSuperWord
-    return vloop().vtrace().is_trace(TraceAutoVectorizationTag::SW_TYPES);
+  bool reduction(Node* n1, Node* n2) const {
+    return vloop_analyzer().reductions().is_marked_reduction_pair(n1, n2);
+  }
+
+  // VLoopMemorySlices Accessors
+  bool same_memory_slice(MemNode* n1, MemNode* n2) const {
+    return vloop_analyzer().memory_slices().same_memory_slice(n1, n2);
+  }
+
+  // VLoopBody Accessors
+  const GrowableArray<Node*>& body() const {
+    return vloop_analyzer().body().body();
+  }
+
+  int bb_idx(const Node* n) const     {
+    return vloop_analyzer().body().bb_idx(n);
+  }
+
+  // VLoopTypes Accessors
+  const Type* velt_type(Node* n) const {
+    return vloop_analyzer().types().velt_type(n);
+  }
+
+  BasicType velt_basic_type(Node* n) const {
+    return vloop_analyzer().types().velt_basic_type(n);
+  }
+
+  bool same_velt_type(Node* n1, Node* n2) const {
+    return vloop_analyzer().types().same_velt_type(n1, n2);
+  }
+
+  int data_size(Node* n) const {
+    return vloop_analyzer().types().data_size(n);
   }
 
+  int vector_width(Node* n) const {
+    return vloop_analyzer().types().vector_width(n);
+  }
+
+  int vector_width_in_bytes(const Node* n) const {
+    return vloop_analyzer().types().vector_width_in_bytes(n);
+  }
+
+#ifndef PRODUCT
+  // TraceAutoVectorization and TraceSuperWord
   bool is_trace_superword_alignment() const {
     // Too verbose for TraceSuperWord
     return vloop().vtrace().is_trace(TraceAutoVectorizationTag::SW_ALIGNMENT);
   }
 
-  bool is_trace_superword_memory_slices() const {
-    return TraceSuperWord ||
-           vloop().vtrace().is_trace(TraceAutoVectorizationTag::SW_MEMORY_SLICES);
-  }
```

Comment:
Note: replaced with `vloop().is_trace_memory_slices()` since all the code is now in `VLoopMemorySlices`.
Tag is renamed to `MEMORY_SLICES`.